### PR TITLE
Remove `decomp_depth` from `qml.device` constructor

### PR DESCRIPTION
**Pull request by albi3ro**
The `decomp_depth` keyword argument only served to prevent operations from being fully decomposed to the target gateset.  It is now removed.

[sc-72706]

### CODE REVIEW

File: pennylane/devices/device_constructor.py [hunk 1]
```diff
@@ -135,10 +127,10 @@ def circuit():
...
 
         def circuit():
-           qml.Hadamard(wires='q11')
-           qml.Hadamard(wires=['ancilla'])
-           qml.CNOT(wires=['q12', -1])
-           ...
+`           qml.Hadamard(wires='q11')
+            qml.Hadamard(wires=['ancilla'])
+            qml.CNOT(wires=['q12', -1])
+            ...`
```
Comment (albi3ro) [inline pennylane/devices/device_constructor.py:133]:
```suggestion
            qml.Hadamard(wires='q11')
            qml.Hadamard(wires=['ancilla'])
            qml.CNOT(wires=['q12', -1])
            ...
```

File: pennylane/devices/device_constructor.py [hunk 2]
```diff
@@ -203,12 +195,12 @@ def ion_trap_cnot(wires, **_):
...
         )
 
-        @qml.qnode(dev)
+        @qml.qnode(dev, expansion_strategy="device")
         def run_cnot():
             qml.CNOT(wires=[0, 1])
             return qml.expval(qml.X(1))
 
-    >>> print(qml.draw(run_cnot, level="device")())
+    >>> print(qml.draw(run_cnot)())
```
Comment (albi3ro) [inline pennylane/devices/device_constructor.py:203]:
```suggestion
    >>> print(qml.draw(run_cnot, level="device")())
```

File: pennylane/devices/device_constructor.py [hunk 3]
```diff
@@ -203,12 +195,12 @@ def ion_trap_cnot(wires, **_):
             'default.qubit', wires=2, custom_decomps={"CNOT" : ion_trap_cnot}
         )
 
-        @qml.qnode(dev)
+        @qml.qnode(dev, expansion_strategy="device")
```
Comment (albi3ro) [inline pennylane/devices/device_constructor.py:198]:
```suggestion
        @qml.qnode(dev)
```

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -39,6 +39,8 @@
   Please use `qml.transforms.split_non_commuting` instead.
   [(#6204)](https://github.com/PennyLaneAI/pennylane/pull/6204)
 
+* The `decomp_depth` keyword argument to `qml.device` is removed.
```
Comment (astralcai) [inline doc/releases/changelog-dev.md:42]:
```suggestion
* The `decomp_depth` keyword argument to `qml.device` is removed.
  [(#6234)](https://github.com/PennyLaneAI/pennylane/pull/6234)
```

**Review by astralcai — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]
Small change needed for rendering docs correctly, otherwise looks good.

**Pull request merged**